lua/lspconfig/rust_analyzer.lua

```diff
@@ -16,15 +16,27 @@ configs.rust_analyzer = {
     cmd = {"rust-analyzer"};
     filetypes = {"rust"};
     root_dir = function(fname)
-      local cargo_metadata = vim.fn.system("cargo metadata --no-deps --format-version 1")
-      local cargo_root = nil
+      -- Order of preference for how we choose the root dir:
+      --   * Current Cargo workspace (if any)
+      --   * Current Cargo crate (if any)
+      --   * Rust project root, for projects that don't use Cargo (if any)
+      --   * Current git repository
+      local cargo_crate_dir = util.root_pattern("Cargo.toml")(fname)
```

Comment:
I would lazily call this to avoid any impact to startup time, the way we have it now short-circuits.